unison-cli/src/Unison/CommandLine/Main.hs

```diff
@@ -98,10 +99,32 @@ main ::
   Codebase IO Symbol Ann ->
   Maybe Server.BaseUrl ->
   UCMVersion ->
-  ((Branch IO, Path.Absolute) -> IO ()) ->
+  (Branch IO -> STM ()) ->
+  (Path.Absolute -> STM ()) ->
   IO ()
-main dir welcome initialPath (config, cancelConfig) initialInputs runtime sbRuntime codebase serverBaseUrl ucmVersion notifyChange = do
-  root <- Codebase.getRootBranch codebase
+main dir welcome initialPath (config, cancelConfig) initialInputs runtime sbRuntime codebase serverBaseUrl ucmVersion notifyBranchChange notifyPathChange = Ki.scoped \scope -> do
+  rootVar <- newEmptyTMVarIO
+  initialRootCausalHash <- Codebase.getRootCausalHash codebase
+  _ <- Ki.fork scope $ do
+    root <- Codebase.getRootBranch codebase
+    atomically $ do
+      -- Try putting the root, but if someone else as already written over the root, don't
+      -- overwrite it.
+      void $ tryPutTMVar rootVar root
+    -- Start forcing the thunk in a background thread.
+    -- This might be overly aggressive, maybe we should just evaluate the top level but avoid
+    -- recursive "deep*" things.
+    void $ UnliftIO.evaluate root
```

Comment:
It's not totally clear to me why we want to force a `Branch` to WHNF. What computations are we hoping to perform eagerly by doing that?

Reply:
Good question, maybe this doesn't actually provide any benefit, but it's tricky to test and be sure 🤷🏼‍♂️ .  It might help force some small computations, and doesn't hurt, so I don't see any reason not to do it, but I can see how that's not super convincing.

Would you like me to remove it?

Reply:
I think it's probably better to remove, then, but no big deal if it stays.